### PR TITLE
Give test service principal admin access to synapse test resources

### LINKED ISSUE #24423: Fix Synapse AccessControl CreateRoleAssignment test
This test currently fails with the following exception:

```
﻿ CreateRoleAssignment
   Source: AccessControlClientLiveTests.cs line 102
   Duration: 28.6 sec

  Message: 
Azure.RequestFailedException : The principal 'c4af0aa8-112d-40bf-af9a-817878df4698' does not have the required Synapse RBAC permission to perform this action. Required permission: Action: Microsoft.Synapse/workspaces/read, Scope: workspaces/workspaceannelosynapse/*.
Status: 403 (Forbidden)
ErrorCode: Unauthorized

Content:
{"error":{"code":"Unauthorized","message":"The principal 'c4af0aa8-112d-40bf-af9a-817878df4698' does not have the required Synapse RBAC permission to perform this action. Required permission: Action: Microsoft.Synapse/workspaces/read, Scope: workspaces/workspaceannelosynapse/*."}}

Headers:
Strict-Transport-Security: REDACTED
x-ms-request-id: f32f29ec-907a-46a0-802f-3f7c153c0a9b
x-ms-client-request-id: 06c2aeaa-94a8-4236-9e1b-92e833e29aee
Content-Length: 281
Content-Type: application/json; charset=utf-8
Date: Mon, 04 Oct 2021 21:43:54 GMT
Server: Microsoft-HTTPAPI/2.0


  Stack Trace: 
<ListRoleDefinitionsAsync>d__10.MoveNext() line 129
--- End of stack trace from previous location where exception was thrown ---
ExceptionDispatchInfo.Throw()
TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
TaskAwaiter`1.GetResult()
<<WrapAsyncResultCore>g__Await|0>d.MoveNext() line 91
--- End of stack trace from previous location where exception was thrown ---
ExceptionDispatchInfo.Throw()
TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
ValueTask`1.get_Result()
<22 more frames...>
ExceptionDispatchInfo.Throw()
TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
TaskAwaiter`1.GetResult()
GenericAdapter`1.BlockUntilCompleted()
NoMessagePumpStrategy.WaitForCompletion(AwaitAdapter awaiter)
AsyncToSyncAdapter.Await(Func`1 invoke)
TestMethodCommand.RunTestMethod(TestExecutionContext context)
TestMethodCommand.Execute(TestExecutionContext context)
<>c__DisplayClass1_0.<Execute>b__0()
BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)

  Standard Output: 
[14:43:52.086] GetToken 
  method: ClientSecretCredential.GetToken
  scopes: [ https://dev.azuresynapse.net/.default ]
  parentRequestId: 06c2aeaa-94a8-4236-9e1b-92e833e29aee


[14:43:52.09] Request 3df95c36-8808-49fd-96d4-00557892d1bd
  method: POST
  uri: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/oauth2/v2.0/token
  headers: 
    x-client-SKU:REDACTED
    x-client-Ver:REDACTED
    x-client-CPU:REDACTED
    x-client-OS:REDACTED
    x-client-current-telemetry:REDACTED
    x-client-last-telemetry:REDACTED
    x-ms-lib-capability:REDACTED
    client-request-id:REDACTED
    return-client-request-id:REDACTED
    x-app-name:REDACTED
    x-app-ver:REDACTED
    Content-Type:application/x-www-form-urlencoded
    x-ms-client-request-id:3df95c36-8808-49fd-96d4-00557892d1bd
    x-ms-return-client-request-id:true
    User-Agent:azsdk-net-Identity/1.4.0 (.NET Framework 4.8.4300.0; Microsoft Windows 10.0.19042 )
  clientAssembly: Azure.Identity


[14:43:52.332] Response 3df95c36-8808-49fd-96d4-00557892d1bd
  status: 200
  reasonPhrase: OK
  headers: 
    Pragma:no-cache
    Strict-Transport-Security:REDACTED
    X-Content-Type-Options:REDACTED
    client-request-id:REDACTED
    x-ms-request-id:b94b1ac7-3f06-49a2-acae-5a7b35b15c00
    x-ms-ests-server:REDACTED
    x-ms-clitelem:REDACTED
    Cache-Control:no-store, no-cache
    Content-Type:application/json; charset=utf-8
    Expires:-1
    P3P:REDACTED
    Set-Cookie:REDACTED
    Date:Mon, 04 Oct 2021 21:43:53 GMT
    Content-Length:1341
  seconds: 0.2410814


[14:43:52.334] GetTokenSucceeded 
  method: ClientSecretCredential.GetToken
  scopes: [ https://dev.azuresynapse.net/.default ]
  parentRequestId: 06c2aeaa-94a8-4236-9e1b-92e833e29aee
  expiresOn: 2021-10-05T21:43:51.0000000+00:00


[14:43:52.335] Request 06c2aeaa-94a8-4236-9e1b-92e833e29aee
  method: GET
  uri: https://workspaceannelosynapse.dev.azuresynapse.net/roleDefinitions?api-version=REDACTED
  headers: 
    Accept:application/json, text/json
    x-ms-client-request-id:06c2aeaa-94a8-4236-9e1b-92e833e29aee
    x-ms-return-client-request-id:true
    User-Agent:azsdk-net-Analytics.Synapse.AccessControl/1.0.0-alpha.20211004.1 (.NET Framework 4.8.4300.0; Microsoft Windows 10.0.19042 )
    Authorization:REDACTED
  clientAssembly: Azure.Analytics.Synapse.AccessControl


[14:43:53.207] ErrorResponse 06c2aeaa-94a8-4236-9e1b-92e833e29aee
  status: 403
  reasonPhrase: Forbidden
  headers: 
    Strict-Transport-Security:REDACTED
    x-ms-request-id:f32f29ec-907a-46a0-802f-3f7c153c0a9b
    x-ms-client-request-id:06c2aeaa-94a8-4236-9e1b-92e833e29aee
    Content-Length:281
    Content-Type:application/json; charset=utf-8
    Date:Mon, 04 Oct 2021 21:43:54 GMT
    Server:Microsoft-HTTPAPI/2.0
  seconds: 0.8709743

```

### PR DESCRIPTION
**Pull request by annelo-msft**
[This PR](https://github.com/Azure/azure-sdk-for-net/pull/23924/files#diff-c87aeec4653a134d00324186ecc94dc9c68f42a760ef2db255487abcee95db89R401) separated the provisioning service principal from the test application, and so they have different sets of permissions.

Since Synapse uses data plane RBAC instead of management plane RBAC, we need to give admin permissions to the test application separately.

Fixes https://github.com/azure/azure-sdk-for-net/issues/24423

### CODE REVIEW

File: sdk/synapse/test-resources-post.ps1 [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+if ($CI) { exit }
+
+Import-Module Az.Synapse
+
+$roleAssignment = (Get-AzSynapseRoleAssignment -WorkspaceName $DeploymentOutputs.AZURE_SYNAPSE_WORKSPACE_NAME -ObjectId $TestApplicationOid)
+
+if (!$roleAssignment) {
+    New-AzSynapseRoleAssignment `
+        -WorkspaceName $DeploymentOutputs.AZURE_SYNAPSE_WORKSPACE_NAME `
+        -RoleDefinitionName "Synapse Administrator" `
```
Comment (christothes) [inline sdk/synapse/test-resources-post.ps1:24]: Is there any risk of the name changing? If so we could use the RoleDefinitionId instead.

**Review by christothes — APPROVED** [review]

File: sdk/synapse/test-resources-post.ps1 (hunk 1 above)
Comment (annelo-msft) [reply sdk/synapse/test-resources-post.ps1:24]: It's documented here: https://docs.microsoft.com/en-us/azure/synapse-analytics/security/synapse-workspace-synapse-rbac-roles ... I don't see RoleDefinitionIds there - do we know those would be more stable?

**Comment by annelo-msft** [comment]
azp run net - synapse - tests

File: sdk/synapse/test-resources-post.ps1 (hunk 1 above)
Comment (christothes) [reply sdk/synapse/test-resources-post.ps1:24]: I don't know if they are more stable, but I've seen role names change prior to GA in the past. The role definition Id would tend not to change. Probably not worth worrying too much about it.

**Pull request merged**